flink-scala-shell/src/main/scala/org/apache/flink/api/scala/FlinkShell.scala

```diff
@@ -277,8 +277,9 @@ object FlinkShell {
 
     val cluster = clusterDescriptor.deploySessionCluster(clusterSpecification)
 
-    val inetSocketAddress = AkkaUtils.getInetSocketAddressFromAkkaURL(
-      cluster.getClusterConnectionInfo.getAddress)
+    val webMonitorUrl = new URL(cluster.getWebInterfaceURL)
+
+    val inetSocketAddress = new InetSocketAddress(webMonitorUrl.getHost, webMonitorUrl.getPort)
```

Comment:
I guess we should make it consistent with `fetchDeployedYarnClusterInfo`. In fact, I think that not trying to resolve the host name might actually be better because Akka expects that messages arrive with the same hostname which it was configured with at start up time.

Reply:
Make sense, will update